src/main/java/be/seeseemelk/mockbukkit/block/state/BlockStateMock.java

```diff
@@ -293,4 +293,17 @@ public static BlockStateMock mockState(@NotNull Block block)
 			return new BlockStateMock(block);
 		}
 	}
+
+	@Override
+	public boolean equals(Object other) {
+		if(other == null) {return false;}
+		if(!(other instanceof BlockState)) {return false;}
+
+		BlockState otherState = (BlockState) other;
+		if(otherState.getType() != getType()) {return false;}
+		if(otherState.isPlaced() != isPlaced()) {return false;}
+		if(!isPlaced()) {return true;}
+
+		return otherState.getBlock().getLocation().equals(getBlock().getLocation());
+	}
```

Comment:
If you add an `equal` method, you should also add a `hashCode` method.

Reply:
I'll add one.

Reply:
hashCode method added - I also modified the existing tests to test hashCode as well.